src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,51 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode ordered_by) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<decltype(typed_segment), const ReferenceSegment&>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto segment_begin, auto segment_end) {
+        auto sorted_segment_search = detail::SortedSegmentSearch(
```

Comment:
What do we need this namespace for? We could also move SortedSegmentSearch into an anonymous namespace if we don't want to expose it.

Reply:
Initially, we had an extra function that used the `SortedSegmentSearch` internally and was the only way to use it (`SortedSegmentSearch` was in an anonymous namespace). Then we refactored the extra function away and moved `SortedSegmentSearch` into the `detail` namespace, since it is only used in the table scan.

I don't understand how moving it into an anonymous namespace would work, because then we couldn't use it from the table scan, could we? 
I've therefore now moved it out of the `detail::` namespace, hope that works! @mrks